WordPress/gutenberg
# Block Patterns: Load patterns from wordpress.org API

## Description
As part of the WordPress.org pattern directory project, we want to load patterns remotely from WordPress.org — this PR is a first step to get there. It unregisters & removes the existing core patterns, and pulls down the patterns from an API on [wordpress.org/patterns](https://wordpress.org/patterns/). These are the same patterns that were just added in #29973. With this change, patterns can be added or updated without needing to release new WordPress or Gutenberg versions.

Right now this preloads the patterns server-side, so functionally there should be no change to the editing experience. Eventually we'll want to load dynamically, client-side, so that authors can search & filter through all patterns on wordpress.org/patterns. See https://github.com/WordPress/pattern-directory/issues/10

## How has this been tested?

- Add or edit a post
- Open the inserter and look at the non-theme patterns
- They should exist 🙂 
- Try to add a core pattern (ex: "Three columns with offset images")
- It should add correctly

~Note: The patterns are rendered as a classic blocks for now, that will be fixed with #28799 - leaving this PR in draft until then.~ This was fixed.

Note 2: There are two patterns - heading and quote - which currently add `blockTypes` support for the heading and quote blocks, respectively. we don't have a way of flagging that on the pattern directory yet, so they're registered without it. See https://github.com/WordPress/pattern-directory/issues/33#issuecomment-824317557
